### PR TITLE
Exie/1135 try another branch

### PR DESCRIPTION
**Pull request by oeddyo**
address: #1135

Changed localPath: String to localPaths: Iterable[String] per Oscar's comment in the issue.

I'm sure I'm doing stupid things here and there. Could someone kindly review my changes please?

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 1]
```diff
@@ -72,7 +81,13 @@ trait LocalSourceOverride extends SchemedSource {
    * @param sinkMode The mode for handling output conflicts.
    * @returns A tap.
    */
-  def createLocalTap(sinkMode: SinkMode): Tap[_, _, _] = new FileTap(localScheme, localPath, sinkMode)
+  def createLocalTap(sinkMode: SinkMode): Tap[JobConf, _, _] = {
+    val taps = localPaths.map {
+      p: String =>
+        CastFileTap(new FileTap(localScheme, p, sinkMode))
+    }.toList
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: can we do:
```scala
taps match {
  case Nil => throw new InvalidSourceException("LocalPaths is empty")
  case oneTap :: Nil => oneTap
  case many => new ScaldingMultiSourceTap(many)
}
```
This way, we don't use the multisource path, which is new in this case, unless you need the new functionality, protecting users who don't make any change.

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 2]
```diff
@@ -196,11 +214,20 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride {
         }
       }
 
-      case Local(_) => {
-        val file = new java.io.File(localPath)
-        if (!file.exists)
+      case Local(strict) => {
+        val files = localPaths.map{
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: let's use: `localPaths.map { p => new java.io.File(p) }` to be more consistent with scalding's style.

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 3]
```diff
@@ -313,12 +340,19 @@ trait SuccessFileSource extends FileSource {
  * Put another way, this runs a Hadoop tap outside of Hadoop in the Cascading local mode
  */
 trait LocalTapSource extends LocalSourceOverride {
-  override def createLocalTap(sinkMode: SinkMode) = new LocalTap(localPath, hdfsScheme, sinkMode).asInstanceOf[Tap[_, _, _]]
+  override def createLocalTap(sinkMode: SinkMode) = {
+    val taps = localPaths.map {
+      p =>
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: put this on the previous line.

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 4]
```diff
@@ -313,12 +340,19 @@ trait SuccessFileSource extends FileSource {
...
+        new LocalTap(p, hdfsScheme, sinkMode).asInstanceOf[Tap[JobConf, RecordReader[_, _], OutputCollector[_, _]]]
+    }.toSeq
+    new ScaldingMultiSourceTap(taps)
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: let's only use the multisource if there is more than one path.

File: scalding-core/src/main/scala/com/twitter/scalding/TimePathedSource.scala [hunk 5]
```diff
@@ -131,7 +131,10 @@ abstract class TimePathedSource(val pattern: String,
   //Write to the path defined by the end time:
   override def hdfsWritePath = TimePathedSource.writePathFor(pattern, dateRange, tz)
 
-  override def localPath = pattern
+  override def localPaths = patterns
+    .flatMap{ pattern: String =>
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/TimePathedSource.scala]: add space after `p` before `{`

File: scalding-core/src/main/scala/com/twitter/scalding/source/CodecSource.scala [hunk 6]
```diff
@@ -56,9 +56,8 @@ class CodecSource[T] private (val hdfsPaths: Seq[String], val maxFailures: Int =
   lazy val field = new Fields(fieldSym.name)
   val injectionBox = Externalizer(injection andThen BytesWritableCodec.get)
 
-  def localPath = {
-    require(hdfsPaths.size == 1, "Local mode only supports a single path");
-    hdfsPaths(0)
+  def localPaths = {
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/source/CodecSource.scala]: `def localPaths = hdfsPaths` is better here. no `{}`

**Comment by johnynek** [comment]
Thanks for this cleanup.

**Comment by oeddyo** [comment]
Something weird is happening in Travis build...

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 7]
```diff
@@ -64,7 +64,16 @@ abstract class SchemedSource extends Source {
  */
 trait LocalSourceOverride extends SchemedSource {
   /** A path to use for the local tap. */
-  def localPath: String
+  def localPaths: Iterable[String]
+
+  // By default, we write to the last path for local paths
+  def localWritePath = localPaths.last
+
+  object CastFileTap {
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: let's not put this as an inner object. Can we make it a `private[scalding]` object in FileSource.scala?

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 8]
```diff
@@ -196,11 +219,16 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride {
...
 
-      case Local(_) => {
-        val file = new java.io.File(localPath)
-        if (!file.exists)
+      case Local(strict) => {
+        val files = localPaths.map{ p => new java.io.File(p) }
+        if (strict && !files.forall(_.exists)) {
           throw new InvalidSourceException(
-            "[" + this.toString + "] Nothing at path: " + localPath)
+            "[" + this.toString + "] Data is missing from one or more paths in: " +
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala]: why not add:

```scala
 Data is missing from: ${localPaths.filterNot { p => (new java.io.File(p)).exists}}
```
so the user can see exactly what is missing?

**Comment by oeddyo** [comment]
Done. @johnynek

**Comment by johnynek** [comment]
merge when green

**Pull request merged**